src/magic_spell_effect.cpp

```diff
@@ -499,3 +499,17 @@ void spell_effect::translocate( const spell &sp, const Creature &caster,
 {
     tp_list.translocate( spell_effect_area( sp, target, spell_effect_blast, caster, true ) );
 }
+
+void spell_effect::use_action( const spell &sp )
+{
+    item granted( sp.effect_data(), calendar::turn );
+    if( !g->u.can_use( granted ) ) {
+        granted.delete();
+        msg = _("You cannot use this spell for that purpose.");
+        return;
+    } else {
+        g->u.i_add( granted );
```

Comment:
If you add it, you should remove it later. But look out of items that get destroyed upon usage.

Anyway, much simpler to just invoke its usage:
```C++
g->u.invoke_item( &granted, g->u.pos() );
````